### DISCUSSION TITLE
페이스북 개발자는 어떻게 일을 할까? 라는 글이 있어서 공유해봅니다 :)

### DISCUSSION THREAD

**Discussion by TaeHyoungKwon** [ThinkAboutSoftware/AcademicConference#171] [General]
[페이스북 개발자는 어떻게 일을 할까?](https://blog.shiren.dev/2022-03-07/)

요런 글이 있길래 공유해봅니다 ㅎㅎ
<img width="3295" alt="페이스북 개발자는 어떻게 일할까" src="https://user-images.githubusercontent.com/18538418/156970170-99d8f29d-67cb-495e-81d1-025f281a23d7.png">

<img width="1222" alt="스크린샷 2022-03-07 오후 1 48 36" src="https://user-images.githubusercontent.com/18538418/156970175-08967bb4-270c-40f3-aea2-bc649bd7869b.png">

**Comment by wooyaggo86** [comment]
너무 좋은 자료에요! 공유해 주시고, 정리도 잘해주셔서 감사합니다👍

**Comment by jongfeel** [comment]
좋은 글 공유 감사합니다.

사실 페이스북이 어떻게 일하는지 관심이 없었다가 읽고 나서 제 의견을 정리해 볼 수 있어서 좋았던 것 같네요.

- What이 주어지면 how는 알아서 => 사실 저는 what도 제가 알아서 찾아서 협업하는 걸 좋아하는 터라 그렇게 하고 있는데, 확실히 how는 숙련도에 따라서 알아서 하는게 맞는 것 같습니다. 그 과정에서 코드리뷰가 포함이 되고 테스트 코드와 같은 원칙들이 지켜지면 품질은 맞춰질테니까요.
- E2E 테스트 및 개발 환경 => 저도 이거 놀랐습니다. 글로만 봐도 놀라운데 실제 경험해 보면 얼마나 놀라울까요? 이 시스템 만드는 개발자도 내부 개발자들일텐데 대단하다는 생각이 듭니다.
- 성과의 기준 네 개 중에, 피플이 눈에 띄었습니다, 피플에서 언급한 내용이 지금 제가 하고 있는 것들인데 아래 항목들입니다.
  - 동료를 도와줌 => 제 메인 업무입니다. 제 업무가 동료를 도와주는 겁니다.
  - 협업을 이렇게 저렇게 잘했음 => 협업의 과정을 통해 업무를 진행합니다. 제 단독으로 진행하는 제품 기여 코드는 없다고 보면 됩니다.
  - 발표를 해서 지식을 공유함 => 발표하는거 좋아하는데 최근 test code & coverage에 관련된 발표를 했었습니다.
  - 위키에 중요 or 복잡한 이슈를 정리해서 공유함 => 어제 singleton 패턴에 대한 의견과 대체 방법에 대한 블로그 글을 작성했습니다.
  - 인터뷰 참여 => 요즘 이직 시즌이 아니라 지원자가 뜸한테 제 업무 중에 일부입니다.
  - 멘토링 => 역량 성장 목표로 많은 분들과 어떻게 하면 성장 목표를 실행하고 달성할지 얘기합니다.
- 마지막 성과 보고서를 어필해야 한다는 점에서, 성과지향적인 평가라는 느낌을 많이 받았습니다. 한국 회사들은 오히려 만족할 만한 성과 평가 기준이 없다 보니 연봉이 올라도 불만인 경우가 있는데, 이런 점이 개선되어야 하지 않나 라는 생각을 해봅니다.
